packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -0,0 +1,327 @@
+ *
+ * @param args - runtime arguments
+ * @returns Name of the YAML file
+ */
+export const getYamlFile = (args: string[]): string => {
+    let yaml = 'ui5.yaml';
+    const index = args.indexOf('--config') !== -1 ? args.indexOf('--config') : args.indexOf('-c');
+
+    if (index !== -1) {
+        yaml = args[index + 1];
+    }
+
+    return yaml;
+};
+
+/**
+ * Gets the path to the webapp folder from the YAML file.
+ *
+ * @param ui5YamlPath - path to the yaml file
+ * @returns Path to the webapp folder
+ */
+export const getWebAppFolderFromYaml = async (ui5YamlPath: string): Promise<string> => {
+    let webAppFolder = 'webapp';
+
+    if (existsSync(ui5YamlPath)) {
+        const ui5Yaml = parseDocument(await promises.readFile(ui5YamlPath, { encoding: 'utf8' })).toJSON();
+
+        if (ui5Yaml.resources?.configuration?.paths?.webapp) {
+            webAppFolder = ui5Yaml.resources.configuration.paths.webapp;
+        }
+    }
+
+    return webAppFolder;
+};
+
+/**
+ * Sends HTML content as a response.
+ *
+ * @param res - The http response object
+ * @param html - The HTML content
+ */
+export const setHtmlResponse = (res: any, html: string): void => {
+    if (res['_livereload']) {
+        res.write(html);
+        res.end();
+    } else {
+        res.status(200).contentType('html').send(html);
+    }
+};
+
+/**
+ * Gets the manifest.json for a given application.
+ *
+ * @param args list of runtime arguments
+ * @returns The content of the manifest.json
+ */
+export const getManifest = async (args: string[]): Promise<SAPJSONSchemaForWebApplicationManifestFile> => {
+    const projectRoot = process.cwd();
+    const yamlFileName = getYamlFile(args);
+    const ui5YamlPath = join(projectRoot, yamlFileName);
+    const webAppFolder = await getWebAppFolderFromYaml(ui5YamlPath);
+    const manifestPath = join(projectRoot, webAppFolder, 'manifest.json');
+    const manifest: SAPJSONSchemaForWebApplicationManifestFile = JSON.parse(
+        await promises.readFile(manifestPath, { encoding: 'utf8' })
+    );
+
+    return manifest;
+};
+
+/**
+ * Gets the minUI5Version from the manifest.json.
+ *
+ * @param args list of runtime args
+ * @returns The minUI5Version from manifest.json or undefined otherwise
+ */
+export async function getUI5VersionFromManifest(args: string[]): Promise<string | undefined> {
+    const manifest: SAPJSONSchemaForWebApplicationManifestFile = await getManifest(args);
+    return manifest['sap.ui5']?.dependencies?.minUI5Version;
+}
+
+/**
+ * Determines which UI5 version to use when previewing the application.
+ *
+ * @param version ui5 version as defined in the yaml or via cli argument
+ * @param log logger for outputing information from where ui5 version config is coming
+ * @returns The UI5 version with which the application will be started
+ */
+export async function setUI5Version(version: string | undefined, log?: ToolsLogger): Promise<string> {
```

Comment:
Why is it called `setUI5Version()`? Looks like this function tries to 'get' or 'resolve' the UI5 version.

Reply:
I didn't came up with a better name. You are right this is not a setter. Will rename to `resolveUI5Version`